### PR TITLE
set-ip-eni-sshkey

### PR DESCRIPTION
**Pull request by ms-petrugarstea**
Code refactoring:
Added set instance eni feature
Added set instance ip address feature
Changed image ssh key dependency from True to False

### CODE REVIEW

File: lib/python/treadmill_aws/cli/admin/aws/image.py [hunk 1]
```diff
@@ -117,13 +117,13 @@ def configure(account, image):
     )
     @click.option(
         '--key',
-        required=True,
+        required=False,
```
Comment (bretttegartms) [inline lib/python/treadmill_aws/cli/admin/aws/image.py]: I'd remove this line; default is False.

File: lib/python/treadmill_aws/cli/admin/aws/instance.py [hunk 2]
```diff
@@ -122,10 +122,20 @@ def configure(instance):
...
     )
+    @click.option(
+        '--ip-address',
+        required=False,
+        help='IP address',
+    )
+    @click.option(
+        '--eni',
+        required=False,
+        help='Network id',
```
Comment (bretttegartms) [inline lib/python/treadmill_aws/cli/admin/aws/instance.py]: Can you give an example of the format you expect the command to be in? 'Network ID' is vague.
e.g. eni-xxxxxxxx

Comment (ms-petrugarstea) [reply lib/python/treadmill_aws/cli/admin/aws/instance.py]: treadmill --dns-domain DOMAIN admin aws instance create --image AMI_ID --eni ENI_ID --size INSTANCE_TYPE --count 1 --role ROLE --hostname HOSTNAME

File: lib/python/treadmill_aws/cli/admin/aws/image.py (hunk 1 above)
Comment (ms-petrugarstea) [reply lib/python/treadmill_aws/cli/admin/aws/image.py]: removed

File: lib/python/treadmill_aws/cli/admin/aws/instance.py (hunk 2 above)
Comment (bretttegartms) [reply lib/python/treadmill_aws/cli/admin/aws/instance.py]: That's not what I meant. "ENI" is an acronym that not everyone may recognize immediately. 
In the **--help=** text, please give an example of the format you expect the user to input. 
e.g. help='Elastic Network ID; e.g. eni-xxxxxxxx' gives a better indication of what the flag does and what we expect for input.

File: lib/python/treadmill_aws/cli/admin/aws/image.py (hunk 1 above)
Comment (bretttegartms) [reply lib/python/treadmill_aws/cli/admin/aws/image.py]: Can you remove the other required=False flags from the file?

File: lib/python/treadmill_aws/cli/admin/aws/instance.py (hunk 2 above)
Comment (ms-petrugarstea) [reply lib/python/treadmill_aws/cli/admin/aws/instance.py]: totally agree. The change has been made.

**Pull request merged**